MaxText/decode.py

```diff
@@ -134,14 +134,14 @@ def decode_loop(config, state=None):
   # Model and Optimizer definition
   model = Transformer(config, mesh = mesh)
 
-  tx = optax.adamw(
-    max_utils.create_learning_rate_schedule(config)
-  ) # TODO: we need an optax.GradientTransformation to form a TrainState, but we don't use it when decoding
-
-
   _, sp_tokenizer = create_data_iterator_with_tokenizer(config, mesh)
 
-  state, state_mesh_annotations = max_utils.setup_initial_state(model, tx, config, rng, mesh, checkpoint_manager)
+  state, state_mesh_annotations = max_utils.setup_initial_state(
```

Comment:
Why don't we have "setup_state_for_inference" and "setup_initial_state" and only pass the right things into the two of them. Going through one interface for the two difference pretty different tasks seems unncessary.

Reply:
Done.